lib/rom/http/dataset.rb

```diff
@@ -124,13 +124,11 @@ class Dataset
       #
       # @api public
       def uri
-        uri = URI(join_path(super, path))
-
-        if get? && params.any?
-          uri.query = param_encoder.call(params)
+        URI(join_path(super, path)).tap do |uri|
+          if params.any?
+            uri.query = param_encoder.call(params)
```

Comment:
If I'm not mistaken, this will add all params as query params, even in a POST request, which could be bad if posting a blob of text for example, and is probably not desired in most scenarios. Perhaps it would be better to add a `query_params` option to `Dataset`?

Reply:
@AMHOL Oh, right. :)  It so happens that the app I am cutting my rom-rb teeth on uses the request body to POST JSON data. I can amend this as you suggest.

Reply:
@AMHOL On further thought (what a day!) the use of both `query_params` *and* `params` seems ... confusing.  I wonder if it would be too complicated to just POST params in the body if there is no other request data?  E.g., in my app I have a `request_data` string var `https://github.com/dchandekstark/solrbee/blob/main/lib/rom/solr/dataset.rb#L12` which gets dumped into the request body.  I could see dumping the params into the body if request_data is nil or empty.

Reply:
Yeah it could be confusing, I'm not sure of a better way to go about it myself, I like the idea of just having params sent as the request body though and having a separate `query_params` option does solve that, perhaps we could have a `query_params` option as well as a `body_params` option, probably best to wait for @solnic to weigh in on this though.

Reply:
I like the idea of `query_params` and `body_params`, it would make things much more obvious.

Reply:
https://github.com/rom-rb/rom-http/pull/44 @solnic @dchandekstark